pkg/asset/store/store.go

```diff
@@ -83,6 +85,19 @@ func (s *storeImpl) Fetch(a asset.Asset, preserved ...asset.WritableAsset) error
 	if wa, ok := a.(asset.WritableAsset); ok {
 		return errors.Wrap(s.purge(append(preserved, wa)), "failed to purge asset")
 	}
+	// On the s390x/s390 platform, the igniton-config file will be converted to config
+	// drive image, so that the terraform can use it to inject metadata.
+	if runtime.GOARCH == "s390x" || runtime.GOARCH == "s390" && a.Config.Libvirt != nil {
+		if reflect.TypeOf(a).String() == "*bootstrap.Bootstrap" ||
+			reflect.TypeOf(a).String() == "*machine.Master" ||
+			reflect.TypeOf(a).String() == "*machine.Worker" {
+			// TODO: convert reflect.TypeOf(a).String() to filename
+			err := configDriveImage(reflect.TypeOf(a).String())
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
this is much like a hacking, do we get other ways? just curious...

Reply:
Yes, I'm trying other ways.

Reply:
I know that we are targeting this change specific to s390x, but would it be more acceptable to target this change for all remote libvirt installations where fw_cfg is not available?
I'm not sure how you make a test for that.

Reply:
Prashanth wanted me to note that we should be careful to ensure this doesn't run when the install type is OpenStack, since it already has an config drive, but this looks specific to libvirt.

Reply:
@jaypoulz Thanks, I think it's a good thought. However, before we use config drive, a configdrive iso should be created, so that terraform can use. The iso file should be created by installer.

> Prashanth wanted me to note that we should be careful to ensure this doesn't run when the install type is OpenStack, since it already has an config drive, but this looks specific to libvirt.

Yes, it effects only when install type is libvirt.